### ISSUE TITLE
`deno.json` is being parsed as `jsonc`

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno#20681]
We should only parse `deno.jsonc` as JSONC. Right now we also parse `deno.json` as JSONC.

```jsonc
// deno.json
{
  tasks: {
    test: "deno repl"
  }
}
```

`deno task test` works

**Comment by dsherret** [comment]
Why not bother to just parse it as jsonc?

**Comment by lucacasonato** [comment]
Because then users can't make the assumption that deno.json is safe for importing using json import attributes / `JSON.parse`

**Comment by dsherret** [comment]
For imports, I think in most cases the user will have control over the deno.json in that case, so when they get an error they can then run it through `deno fmt` or manual fix any non-JSON issues.

For JSON.parse, that might occur with libraries parsing a user's deno.json, but if they want to support modifying deno.jsonc they will have to deal with the same issue of not being able to use JSON.parse. Plus it will error in that scenario, so the user will easily see it errors then update their deno.json.

Basically, I don't see the harm in us just treating them the same and not erroring. Plus this would be a breaking change.
